REST API v2.1.0 documentation

abellotti opened the pull request:
REST API v2.1.0 documentation.

@mfeifer please review.

Marked as WIP as I want to squash commits before merging.  Just thought it may be easier to review.

Fryguy commented on api/appendices/provider_types.adoc:
```diff
@@ -0,0 +1,50 @@
+
```
I thought leading blanks in the files caused problems

abellotti replied:
actually, the leading blanks in the files fix the problem asciibinder had with the [[ ]] tagged entries where they were shown in the rendered html.

imtayadeway commented on api/appendices/resource_attributes.adoc:
```diff
@@ -0,0 +1,232 @@
+| category | hash | Category identifier for which the tag is getting added to, requires an *href*, *id* or *name* attribute in the hash
+|=====================
+
+When adding via the *assign* action to /api/categories/:id/tags, the category should not be specified
```
Not sure I understand this. Is it not the `create` action?

imtayadeway commented on api/overview/auth.adoc:
```diff
@@ -67,8 +66,8 @@ Content-Type: application/json
-       -H "X-Auth-Token: af0245-238722-4d23db"
-       http://localhost:3000/api/services/1013
+       -H "X-Auth-Token: 93d98fd44cc5a401ff385ba87b21fd13"
+       http://localhost:300/api/services/1013
```
Is this really port 300?

imtayadeway commented on api/overview/auth.adoc:
```diff
@@ -85,3 +84,50 @@ When a request fails due to an invalid token, the client would then need
+----
+
+[[api-token-ttl-configuration]]
+==== Configuing the REST API Token Duration
```
s/Configuing/Configuring/

imtayadeway commented on api/overview/authorization.adoc:
```diff
@@ -0,0 +1,72 @@
+When authenticating to the appliance, the current group of the authenticated user will be used for
+determining the authorization of what the user can see and do using the REST API. While this
+is acceptable when users only belong to a single group, this falls short when users are authenticating
+using their External Authentication credentials (IPA, Ldap or AD) where they may belong to
```
Not sure, but should Ldap be uppercase?

imtayadeway commented on api/reference/categories_tags.adoc:
```diff
@@ -0,0 +1,430 @@
+[[creating-tags]]
+=== Creating Tags
+
+New tags for a category can be created as either as a tag collection post or via a tag
```
maybe simply "[...] created either as a tag [...]"

imtayadeway commented on api/reference/categories_tags.adoc:
```diff
@@ -0,0 +1,430 @@
+[[editing-categories-tags]]
+=== Editing Categories & Tags
+
+Deleting Categories and Tags can be done via the *edit* action:
```
s/Deleting/Editing/

imtayadeway commented on api/reference/service_custom_actions.adoc:
```diff
@@ -0,0 +1,442 @@
+This is made available with the _custom_actions_ and _custom_action_buttons_ virtual attributes on service_templates
+and services.
+
+* _custom_actions_ return the custom buttons as layed out in the UI on the service. The information returned includes
```
s/return/returns/

imtayadeway commented:
@abellotti this is awesome :fireworks: . Minor comments above

abellotti commented:
Thanks Tim for the review, I'll update and repush.

abellotti commented:
Updated and moved examples to separate topic on left hand side for easier browsing of examples by category.

@imtayadeway @mfeifer Let me know if there's anything else to update, if not, I'll squash/repush/unwip.  Thanks.

imtayadeway commented:
@abellotti :+1: :cake: :musical_keyboard:

abellotti commented:
@mfeifer anything to change ? can this be merged ?  Thanks.
